app/src/main/java/com/duckduckgo/app/settings/db/SettingsDataStore.kt

```diff
@@ -19,19 +19,19 @@ package com.duckduckgo.app.settings.db
 import android.content.Context
 import android.content.SharedPreferences
 import androidx.core.content.edit
+import com.duckduckgo.app.global.DuckDuckGoTheme
 import javax.inject.Inject
 
-
 interface SettingsDataStore {
-    var lightThemeEnabled: Boolean
+    var theme: DuckDuckGoTheme?
     var autoCompleteSuggestionsEnabled: Boolean
 }
 
 class SettingsSharedPreferences @Inject constructor(private val context: Context) : SettingsDataStore {
 
-    override var lightThemeEnabled: Boolean
-        get() = preferences.getBoolean(KEY_LIGHT_THEME_ENABLED, false)
-        set(enabled) = preferences.edit { putBoolean(KEY_LIGHT_THEME_ENABLED, enabled) }
+    override var theme: DuckDuckGoTheme?
+        get() = preferences.getString(KEY_THEME, null)?.let { DuckDuckGoTheme.valueOf(it) }
```

Comment:
This line feels like it's doing a lot, and it makes it hard to parse quickly.

Reply:
That's slightly better, though I'd still ultimately prefer a more verbose `if` I think. (teeny, tiny preference though! 😄)